tests/test_lists2safebrowsing.py

```diff
@@ -168,3 +215,279 @@ def test_add_domain_to_list_duplicate():
+
+    assert output_writes == [call(expected_output)]
+    assert capsys.readouterr().out == expected_print
+
+
+def test_write_safebrowsing_blocklist_no_test_domains(capsys, chunknum):
+    """Test write_safebrowsing_blocklist without adding test domains."""
+    # Store reference to original function before mocking it
+    original_add_domain_to_list = l2s.add_domain_to_list
+
+    # Only mock the first two calls, which add the test domains
+    def side_effect(*args, **kwargs):
+        side_effect.call_counter += 1
+        if side_effect.call_counter < 3:
+            return False
+        return original_add_domain_to_list(*args, **kwargs)
+
+    side_effect.call_counter = 0
+
+    with patch("lists2safebrowsing.add_domain_to_list") as m:
+        m.side_effect = side_effect
+        output_writes, expected_output, expected_print = \
+            _write_safebrowsing_blocklist(chunknum, "78.0", False)
+
+    assert output_writes == [call(expected_output)]
+    assert capsys.readouterr().out == expected_print
+
+
+def test_write_safebrowsing_blocklist_no_output_file(capsys, chunknum):
+    """Test write_safebrowsing_blocklist without an output file."""
+    _, _, expected_print = _write_safebrowsing_blocklist(chunknum, "78.0",
+                                                         write_to_file=False)
+    assert capsys.readouterr().out == expected_print
+
+
+def _get_expected_results(list_type, domains, chunknum, section):
+    """Prepare expected results for entity/plugin list generation tests."""
+    if list_type == LIST_TYPE_ENTITY:
+        canonicalized_domains = [l2s.canonicalize(d[1]) for d in domains]
+        domains = [d[0] for d in domains]
+        log_id = "entity"
+        print_id = "Entity list"
+    else:
+        canonicalized_domains = [l2s.canonicalize(d) for d in domains]
+        log_id = "plugin-blocklist"
+        print_id = "Plugin blocklist"
+
+    # Remove duplicates while preserving order
+    expected_domains = OrderedDict()
+    for d, c_d in zip(domains, canonicalized_domains):
+        if c_d not in expected_domains:
+            expected_domains[c_d] = d
+    expected_hashes = [hashlib.sha256(d.encode("utf-8"))
+                       for d in expected_domains]
+    expected_bytes = hashlib.sha256().digest_size * len(expected_hashes)
+    expected_header = b"a:%d:32:%d\n" % (chunknum, expected_bytes)
+    expected_output_writes = ([call(expected_header)]
+                              + [call(h.digest()) for h in expected_hashes])
+
+    expected_log_writes = [
+        call(LOG_MSG % (log_id, d, c_d, h.hexdigest()))
+        for (c_d, d), h in zip(expected_domains.items(), expected_hashes)
+    ]
+    # FIXME: os.fstat returns 0 size for the mocked file. Should be:
+    # len(expected_header) + expected_bytes
+    expected_print = (PRINT_MSG % (print_id, section,
+                      len(expected_domains), 0))
+
+    return expected_output_writes, expected_log_writes, expected_print
```

Comment:
When testing the expected values in the unittest we should be comparing these values with static file or static expected values that we can compare to rather than generating one. Generally, unittests should be simple and repetitive. Generating the expected result using a function like this results into duplicated logic and extra maintenance whenever we change the functions we are testing. Can you upload and compare the expected values to the respective tests?

Reply:
Done! I changed the tests to use static expected values instead of generating them. Thanks for suggesting this!